java/src/shared/definitions/TurnStatus.java

```diff
@@ -183,6 +183,12 @@ public TurnResult endTurn(@NotNull TurnTracker tt, @NotNull Player p) {
         if (!canEndTurn(tt, p)) {
             return null;
         }
+        for (DevCardType d : DevCardType.values()) {
```

Comment:
This should be in `PLAYING.advanceTurn`. Always go for the most granular possible.